odl/discr/lp_discr.py

```diff
@@ -446,8 +528,16 @@ def __setitem__(self, indices, values):
             shape is allowed as ``values``.
         """
         if values in self.space:
-            self.ntuple.__setitem__(indices, values.ntuple)
+            # For RawDiscretizationVector of the same type, use ntuple directly
+            return self.ntuple.__setitem__(indices, values.ntuple)
+        elif (isinstance(self.space.dspace, Ntuples) and
+              equiv_views(self.asarray(), values) and
```

Comment:
I can add a TODO, we probably want to use a context manager once that is in place. This was more like a quick fix.